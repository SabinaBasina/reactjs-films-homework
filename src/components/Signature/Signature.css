.about {
  background-color: rgba(0, 0, 0, 0.753);
}

.text {
  color: white;
}
