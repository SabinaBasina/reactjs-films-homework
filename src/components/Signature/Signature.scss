$bg-color: rgb(252, 141, 159);

h1 {
  color: $bg-color;
}

.about {
  background-color: rgba(0, 0, 0, 0.753);
}

.text {
  color: white;
}
