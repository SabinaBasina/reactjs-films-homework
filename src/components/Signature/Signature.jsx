import React, { Component } from 'react';
import PropTypes from 'prop-types';
<<<<<<< HEAD
import styles from './Signature.scss';


class Signature extends Component {
  render() {
    const { name } = this.props;
    return (
      <main>
        <h1>
          Hello,
          {name}
        </h1>
        <article className={styles.about}>
          <p className={styles.text}>
            Text1
          </p>
        </article>
      </main>
=======
import './Signature.scss';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import Home from '../Home/Home.jsx';
import Menu from '../Menu/Menu.jsx';
import TvShowDetails from '../TvShowDetails/TvShowDetails.jsx';

class Signature extends Component {
  render() {
    return (
      <Router>
        <div>

          <Menu />

          <Route path="/" exact component={Home} />
          <Route
            path="/tvShowDetails/:id"
            render={props => <TvShowDetails id={props.match.params.id} />}
          />

        </div>
      </Router>
>>>>>>> origin/part2
    );
  }
}

Signature.propTypes = {
  name: PropTypes.string,
<<<<<<< HEAD
};

Signature.defaultProps = {
  name: 'Name',
=======
>>>>>>> origin/part2
};

export default Signature;
