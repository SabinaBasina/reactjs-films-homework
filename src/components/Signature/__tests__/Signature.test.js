import React from 'react';
import Signature from '../Signature.jsx';
//import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

test('Signature test', () => {
// const component = renderer.create(
//     <Signature name='Name'></Signature>,
// );
// let tree = component.toJSON();
  const renderer = new ShallowRenderer();
  renderer.render(<Signature />);
  const result = renderer.getRenderOutput();

    
  expect(result).toMatchSnapshot();
});

