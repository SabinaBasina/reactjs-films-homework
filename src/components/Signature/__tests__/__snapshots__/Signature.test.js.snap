// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature test 1`] = `
<<<<<<< HEAD
<main>
  <h1>
    Hello,
    Name
  </h1>
  <article
    className="about"
  >
    <p
      className="text"
    >
      Text1
    </p>
  </article>
</main>
=======
<BrowserRouter>
  <div>
    <Menu />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      path="/tvShowDetails/:id"
      render={[Function]}
    />
  </div>
</BrowserRouter>
>>>>>>> origin/part2
`;
